src/client.ts

```diff
@@ -55,8 +55,37 @@ export default class UnleashClient extends EventEmitter {
     }
   }
 
+  isParentDependencySatisfied(feature: FeatureInterface | undefined, context: Context) {
+    if (!feature?.name) {
+      return true;
+    }
+
+    if (feature.dependencies && feature.dependencies.length) {
+      return feature.dependencies.every(parent => {
+        const parentFeature = this.repository.getToggle(parent.feature);
+
+        if (parentFeature?.dependencies?.length) {
+          return false;
+        }
+
+        if (parent.enabled !== false && parent.variants?.length) {
+          return parent.variants.includes(this.getVariant(parent.feature, context).name);
+        }
+
+        return parent.enabled !== false
```

Comment:
Can we make it more readable, and not do negate operator, but instead `===`

Reply:
by default enabled is not provided so it's undefined. enabled: false is a special case for kill-switches in parent flag.This code is explicit about explicit enabled false and everything else.

Reply:
I reorganized the conditions a bit but the !== false has to stay as it is